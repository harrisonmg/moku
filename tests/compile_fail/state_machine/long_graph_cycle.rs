use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Foo;

    #[superstate(Beef)]
    impl State<BlinkyState> for Foo {}

    struct Bar;

    #[superstate(Foo)]
    impl State<BlinkyState> for Bar {}

    struct Fizz;

    #[superstate(Foo)]
    impl State<BlinkyState> for Fizz {}

    struct Buzz;

    #[superstate(Fizz)]
    impl State<BlinkyState> for Buzz {}

    struct Dead;

    #[superstate(Buzz)]
    impl State<BlinkyState> for Dead {}

    struct Beef;

    #[superstate(Dead)]
    impl State<BlinkyState> for Beef {}
}

fn main() {}
