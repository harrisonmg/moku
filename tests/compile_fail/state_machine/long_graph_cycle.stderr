error: state graph cycle detected in children of Dead:
       Dead
       └─ Beef
          └─ Foo
             ├─ Bar
             └─ Fizz
                └─ Buzz
                   └─ Dead
  --> tests/compile_fail/state_machine/long_graph_cycle.rs:40:18
   |
40 |     #[superstate(Dead)]
   |                  ^^^^
